checker/src/main/java/alluxio/checker/SparkIntegrationChecker.java

```diff
@@ -0,0 +1,265 @@
+
+    // If one executor cannot recognize Alluxio, something wrong has happened
+    boolean canRecognizeClass = true;
+    boolean canRecognizeFS = true;
+    for (Tuple2<Integer, String> op : sSparkJobResult) {
+      if (op._1() == 1) {
+        canRecognizeClass = false;
+      }
+      if (op._1() == 2) {
+        canRecognizeFS = false;
+      }
+    }
+    return canRecognizeClass ? (canRecognizeFS ? 0 : 2) : 1;
+  }
+
+  /**
+   * Checks if this Spark driver or executors can recognize Alluxio classes and filesystem.
+   *
+   * @return 0 succeed, 1 fail to recognize Alluxio classes,
+   * 2 fail to recognize Alluxio filesystem
+   */
+  private static int getStatus() {
+    // Checks if Spark driver or executors can recognize Alluxio classes
+    try {
+      Class.forName("alluxio.AlluxioURI");
+      Class.forName("alluxio.client.file.BaseFileSystem");
+      Class.forName("alluxio.hadoop.AlluxioFileSystem");
+    } catch (ClassNotFoundException e) {
+      return 1;
+    }
+
+    // checks if Spark driver or executors can recognize Alluxio filesystem
+    try {
+      FileSystem.getFileSystemClass("alluxio", new Configuration());
+    } catch (Exception e) {
+      return 2;
+    }
+
+    return 0;
+  }
+
+  /**
+   * Gets the current Spark executor IP address.
+   *
+   * @param s the integer pass in
+   * @return the current Spark executor IP address
+   */
+  private static String getAddress(Integer s) {
+    String address;
+    try {
+      address = InetAddress.getLocalHost().getHostAddress();
+    } catch (UnknownHostException e) {
+      LOG.debug("cannot get host address");
+      address = "unknown address";
+    }
+    return address;
+  }
+
+  /**
+   * Merges the IP addresses that have the same key.
+   *
+   * @param a, b two IP addresses that both can recognize Alluxio or both cannot
+   * @return merged Spark executor IP addresses
+   */
+  private static String merge(String a, String b) {
+    if (a.contains(b)) {
+      return a;
+    } else if (b.contains(a)) {
+      return b;
+    } else {
+      return a + "; " + b;
+    }
+  }
+
+  /**
+   * Prints the user-facing messages.
+   *
+   * @param resultCode the result code get from checkIntegration
+   * @param conf the current SparkConf
+   */
+  private static void printMessage(int resultCode, SparkConf conf) {
+    if (sAlluxioHAMode) {
+      System.out.println("Alluixo is running in high availbility mode.");
+    }
+
+    try {
+      for (Tuple2<Integer, String> sjr : sSparkJobResult) {
+        if (sjr._1() == 1) {
+          System.out.println("IP addresses: " + sjr._2 + " cannot recognize Alluxio classes.");
+        } else if (sjr._1() == 2) {
+          System.out.println("IP addresses: " + sjr._2 + " cannot recognize Alluxio filesystem.");
+        } else {
+          System.out.println("IP addresses: " + sjr._2 + " can recognize Alluxio filesystem.");
+        }
+      }
+    } catch (NullPointerException e) {
+      //
```

Comment:
if ignoring an exception, always provide a comment explaining why it is ok to ignore